partner_contact_in_several_companies/models/res_partner.py

```diff
@@ -0,0 +1,158 @@
+            ("attached", "Attached to existing Contact"),
+        ],
+        compute="_compute_contact_type",
+        store=True,
+        index=True,
+        default="standalone",
+    )
+    contact_id = fields.Many2one(
+        "res.partner",
+        string="Main Contact",
+        domain=[("is_company", "=", False), ("contact_type", "=", "standalone")],
+    )
+    other_contact_ids = fields.One2many(
+        "res.partner",
+        "contact_id",
+        string="Others Positions",
+    )
+
+    @api.depends("contact_id")
+    def _compute_contact_type(self):
+        for rec in self:
+            rec.contact_type = "attached" if rec.contact_id else "standalone"
+
+    def _basecontact_check_context(self, mode):
+        """Remove "search_show_all_positions" for non-search mode.
+        Keeping it in context can result in unexpected behaviour (ex: reading
+        one2many might return wrong result - i.e with "attached contact"
+        removed even if it"s directly linked to a company).
+        Actually, is easier to override a dictionary value to indicate it
+        should be ignored...
+        """
+        if mode != "search" and "search_show_all_positions" in self.env.context:
+            result = self.with_context(search_show_all_positions={"is_set": False})
+        else:
+            result = self
+        return result
+
+    @api.model
+    def search(self, args, offset=0, limit=None, order=None, count=False):
+        """Display only standalone contact matching ``args`` or having
+        attached contact matching ``args``"""
+        ctx = self.env.context
+        if (
+            ctx.get("search_show_all_positions", {}).get("is_set")
+            and not ctx["search_show_all_positions"]["set_value"]
+        ):
+            args = expression.normalize_domain(args)
+            attached_contact_args = expression.AND(
+                (args, [("contact_type", "=", "attached")])
+            )
+            attached_contacts = super(ResPartner, self).search(attached_contact_args)
+            args = expression.OR(
+                (
+                    expression.AND(([("contact_type", "=", "standalone")], args)),
+                    [("other_contact_ids", "in", attached_contacts.ids)],
+                )
+            )
+        return super(ResPartner, self).search(
+            args, offset=offset, limit=limit, order=order, count=count
+        )
+
+    @api.model
+    def create(self, vals):
+        """When creating, use a modified self to alter the context (see
+        comment in _basecontact_check_context).  Also, we need to ensure
+        that the name on an attached contact is the same as the name on the
+        contact it is attached to."""
+        modified_self = self._basecontact_check_context("create")
+        if not vals.get("name") and vals.get("contact_id"):
+            vals["name"] = modified_self.browse(vals["contact_id"]).name
+        return super(ResPartner, modified_self).create(vals)
```

Comment:
**issue**: When I install the module, I have this warning:

```
WARNING devel odoo.api.create: The model odoo.addons.partner_contact_in_several_companies.models.res_partner is not overriding the create method in batch 
```

It's probably enough to just change the decorator to `@api.model_create_multi`